modules/dnn/src/ie_ngraph.cpp

```diff
@@ -0,0 +1,681 @@
+            if (!param_pluginPath.empty())
+            {
+                candidates.push_back(param_pluginPath);
+            }
+            bool found = false;
+            for (size_t i = 0; i != candidates.size(); ++i)
+            {
+                const std::string& libName = candidates[i];
+                try
+                {
+                    InferenceEngine::IExtensionPtr extension =
+                        InferenceEngine::make_so_pointer<InferenceEngine::IExtension>(libName);
+
+                    ie.AddExtension(extension, "CPU");
+                    CV_LOG_INFO(NULL, "DNN-IE: Loaded extension plugin: " << libName);
+                    found = true;
+                    break;
+                }
+                catch(...) {}
+            }
+            if (!found && !candidates.empty())
+            {
+                CV_LOG_WARNING(NULL, "DNN-IE: Can't load extension plugin (extra layers for some networks). Specify path via OPENCV_DNN_IE_EXTRA_PLUGIN_PATH parameter");
+            }
+            // Some of networks can work without a library of extra layers.
+            // OpenCV fallbacks as extensions.
+            ie.AddExtension(std::make_shared<InfEngineExtension>(), "CPU");
+#ifndef _WIN32
+            // Limit the number of CPU threads.
+            if (device_name == "CPU")
+                ie.SetConfig({{
+                    InferenceEngine::PluginConfigParams::KEY_CPU_THREADS_NUM, format("%d", getNumThreads()),
+                }}, device_name);
+#endif
+        }
+        std::map<std::string, std::string> config;
+        if (device_name == "MYRIAD") {
+            config.emplace("VPU_DETECT_NETWORK_BATCH", CONFIG_VALUE(NO));
+        }
+        netExec = ie.LoadNetwork(net, device_name, config);
+    }
+    catch (const std::exception& ex)
+    {
+        CV_Error(Error::StsError, format("Failed to initialize Inference Engine backend (device = %s): %s", device_name.c_str(), ex.what()));
+    }
+}
+
+bool InfEngineNgraphNet::isInitialized()
+{
+    return isInit;
+}
+
+bool NgraphBackendLayer::getMemoryShapes(const std::vector<MatShape> &inputs,
+                                            const int requiredOutputs,
+                                            std::vector<MatShape> &outputs,
+                                            std::vector<MatShape> &internals) const
+{
+    InferenceEngine::ICNNNetwork::InputShapes inShapes = t_net.getInputShapes();
+    InferenceEngine::ICNNNetwork::InputShapes::iterator itr;
+    bool equal_flag = true;
+    size_t i = 0;
+    for (itr = inShapes.begin(); itr != inShapes.end(); ++itr)
+    {
+        InferenceEngine::SizeVector currentInShape(inputs[i].begin(), inputs[i].end());
+        if (itr->second != currentInShape)
+        {
+            itr->second = currentInShape;
+            equal_flag = false;
+        }
+        i++;
+    }
+
+    if (!equal_flag)
+    {
+        InferenceEngine::CNNNetwork curr_t_net(t_net);
+        curr_t_net.reshape(inShapes);
```

Comment:
Why is ".get()" method with "const" logical modifier doing some network reshaping?

Reply:
It looks like a workaround introduced at https://github.com/opencv/opencv/pull/13608 to dynamically reshape networks from OpenVINO's IRs in case of different input shape.